server/service/src/rnr_form/generate_rnr_form_lines.rs

```diff
@@ -133,7 +134,7 @@ pub fn generate_rnr_form_lines(
                 requisition_line_id: None,
                 previous_monthly_consumption_values,
                 average_monthly_consumption,
-                initial_balance,
+                initial_balance: pos_zero(initial_balance),
```

Comment:
Roxy did notice this in earlier work but I couldn't reproduce. I was hoping if we did reproduce we'd fix it at the DB level otherwise anything depending on the source of the issue will have to bodge it.

I suspect `get_opening_balance` which gets the initial balance could only be returning `-0.0` if the previous final balance is negative or `StockOnHandRepository` is returning `-0.0`.

(deduced from in rust `0.0 - 0.0` is `0.0`, and `-0.0 - 0.0` is `-0.0`)